app/models/ability.rb

```diff
@@ -79,6 +80,14 @@ def address_abilities
     can [:create, :show, :destroy], Address, addressable_type: "Contact" if can_manage_delivery_address?
   end
 
+  def beneficiary_abilities
+    can :create, Beneficiary
+    can [:create, :index, :show, :update], Beneficiary, created_by_id: @user_id
+    if can_manage_orders? || @api_user
+      can [:create, :index, :show, :update], Beneficiary
```

Comment:
For `can_manage_orders` and `api_user`, we can probably allow `:destroy` here as well.